### PR TITLE
Adds a general container based development workflow for mixer

### LINKED ISSUE #1060: Tidy up documentation around the workflows for mixer
In PR https://github.com/istio/mixer/pull/1345, general consensus was reached that the build workflow for mixer in https://github.com/istio/mixer/issues/1337 should be submitted to https://github.com/istio/istio/blob/master/devel/README.md.

### PR DESCRIPTION
**Pull request by sdake**
This PR improves the developer documentation slightly to explain how to build container images and update the Istio manifests.

Fixes #1060

**Special notes for your reviewer**:

The review of https://github.com/istio/mixer/pull/1345 resulted in this PR.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by sdake** [comment]
@ldemailly perhaps @mandarjog is traveling or otherwise occupied.  Any chance you can provide a review of this change as it already underwent pretty significant reviews in PR https://github.com/istio/mixer/pull/1345 which you were involved in?  A colleague has been custom building pilot, and wants to submit a dependent change.

Cheers
-steve

File: devel/README.md [hunk 1]
```diff
@@ -255,6 +263,38 @@ tooling functions correctly
...
+* The `-t` parameter `my-tag` is the desired tag. This should be customized.
+
+### Building the Istio manfiests
+
+Use [updateVersion.sh](https://github.com/istio/istio/blob/master/install/updateVersion.sh)
+to generate new manifests with the specified Mixer containers.
+
+```
+cd $ISTIO/istio
+install/updateVersion.sh -r gcr.io/my-project,tag
```
Comment (ldemailly) [inline devel/README.md]: one of these doesn't work well if you put a space (https://github.com/istio/istio/issues/642) are you sure this works as is ?

**Comment by ldemailly** [comment]
/ok-to-test

ps: I'm traveling and in vacation some once you say it's indeed working without adding a space, then someone else needs to  / l g t m it so it gets merged (I may get around to it on friday)

File: devel/README.md (hunk 1 above)
Comment (sdake) [reply devel/README.md]: very odd; thanks for the catch!  i'll update the PR now.

File: devel/README.md [hunk 2]
```diff
@@ -271,6 +279,39 @@ tooling functions correctly
...
+* The `-t` parameter `my-tag` is the desired tag. This should be customized.
+
+### Building the Istio manfiests
+
+Use [updateVersion.sh](https://github.com/istio/istio/blob/master/install/updateVersion.sh)
+to generate new manifests with the specified Mixer containers.
+
+```
+cd $ISTIO/istio
+install/updateVersion.sh -rgcr.io/my-project,tag
```
Comment (ldemailly) [inline devel/README.md]: mixer is -x afaik - did you test (run the command, followed by git diff istio.VERSION ?)
also it's best if you add commits vs force pushing updates so the preview review don't disappear

Comment (sdake) [reply devel/README.md]: @ldemailly I did test with git diff (this is my standard workflow) - however it was a groggy ~2am (my teenagers woke me up...) so missed the fact that this was modifying proxy not mixer.

Re force pushing will do in the future - the current workflow states to force push - however the workflow doesn't address the case of making a change to a PR.  Will do additional commits in the future.

Fix incoming.

Comment (ldemailly) [reply devel/README.md]: oh can you edit wherever it says to force push, that's bad after the PR was once submitted

that's why we use "squash and merge" so in master there will be the final product but the PR should have the iterations

**Comment by sdake** [comment]
@ldemailly you got it - I'll sort out a doc change now.

Cheers
-steve

**Pull request merged**